### PR TITLE
Optimization // Added some system features from Genshin pizza helper

### PR DESCRIPTION
**Pull request by Bill-Haku**
implement update checks, user policy, etc.

### CODE REVIEW

File: Common Tools/TapticSupport.swift
Comment (CanglongCl) [inline Common Tools/TapticSupport.swift:1]: Comment should be in English

File: Common Tools/TapticSupport.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    case error
+    case light
+    case medium
+    case heavy
+    case rigid
+    case soft
+    case selection
+}
+
+func simpleTaptic(type: SimpleTapticType) {
```
Comment (CanglongCl) [inline Common Tools/TapticSupport.swift:26]: SwiftLint: Cyclomatic Complexity Violation: Function should have complexity 10 or less; currently complexity is 12 (cyclomatic_complexity)

File: Common Tools/DeviceRecognizer.swift [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children
+            .reduce("") { identifier, element in
+                guard let value = element.value as? Int8,
+                      value != 0 else { return identifier }
+                return identifier + String(UnicodeScalar(UInt8(value)))
+            }
+
+        // swiftlint:disable:next cyclomatic_complexity
+        func mapToDevice(identifier: String)
```
Comment (CanglongCl) [inline Common Tools/DeviceRecognizer.swift:86]: Superfluous Disable Command Violation: SwiftLint rule 'cyclomatic_complexity' did not trigger a violation in the disabled region; remove the disable command (superfluous_disable_command)

File: Common Tools/DeviceRecognizer.swift [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            case "iPad8,11",
+                 "iPad8,12": return "iPad Pro (12.9-inch) (4th generation)"
+            case "iPad13,8", "iPad13,9", "iPad13,10",
+                 "iPad13,11": return "iPad Pro (12.9-inch) (5th generation)"
+            case "AppleTV5,3": return "Apple TV"
+            case "AppleTV6,2": return "Apple TV 4K"
+            case "AudioAccessory1,1": return "HomePod"
+            case "AudioAccessory5,1": return "HomePod mini"
+            case "i386",
+                 "x86_64": return "Simulator \(mapToDevice(identifier: ProcessInfo().environment["SIMULATOR_MODEL_IDENTIFIER"] ?? "iOS"))"
```
Comment (CanglongCl) [inline Common Tools/DeviceRecognizer.swift:166]: Line Length Violation: Line should be 120 characters or less; currently it has 138 characters (line_length)

File: Packages/HBMihoyoAPI/Sources/HBMihoyoAPI/Common/Models/Server.swift [hunk 4]
```diff
@@ -60,7 +60,19 @@ extension Server: Codable {}
 
 extension Server: CustomStringConvertible {
     public var description: String {
```
Comment (CanglongCl) [inline Packages/HBMihoyoAPI/Sources/HBMihoyoAPI/Common/Models/Server.swift:62]: i18n here? Or just leave it?

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HBPizzaHelperAPI.swift [hunk 5]
```diff
@@ -0,0 +1,299 @@
+import Foundation
+
+@available(iOS 13, watchOS 7, *)
+public enum PizzaHelperAPI {
+    /// 获取最新系统版本
```
[the remark this answers is not in the document]
Comment (CanglongCl) [reply Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HBPizzaHelperAPI.swift:5]: also, use `async`

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HBPizzaHelperAPI.swift [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+                    print("request succeed")
+                    completion(requestResult)
+
+                case .failure:
+                    print("fetch ENCharacterLocDatas fail")
+                }
+            }
+    }
+
+    public static func getArtifactRatingScore(
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HBPizzaHelperAPI.swift:99]: delete

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+    case post
+    case get
+    case put
+}
+
+// MARK: - HttpMethod
+
+@available(iOS 13, watchOS 6, *)
+struct HttpMethod<T: Codable> {
+    /// 返回自己的后台的结果接口
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift:23]: in English

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+    ///   - completion:异步返回处理好的data以及报错的类型
+    ///
+    ///  需要自己传URL类型的url过来
+    static func homeRequest(
+        _ method: Method,
+        _ urlStr: String,
+        cachedPolicy: URLRequest.CachePolicy = .useProtocolCachePolicy,
+        hostType: HostType = .generalHost,
+        keyDecodingStrategy: JSONDecoder.KeyDecodingStrategy = .useDefaultKeys,
+        completion: @escaping (
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift:36]: use `async`

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+    }
+
+    /// 返回需要POST的结果接口
+    /// - Parameters:
+    ///   - method:Method, http方法的类型
+    ///   - url:String，请求的路径
+    ///   - completion:异步返回处理好的data以及报错的类型
+    ///
+    ///  需要自己传URL类型的url过来
+    static func postRequest(
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift:130]: refactor this whole func.
- use `async`
- no need to set header like `x-rpc-client_type` etc
- no need to set `method` here
- delete those `print`
-

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift [hunk 10]
```diff
@@ -0,0 +1,486 @@
...
+                                )))
+                            }
+                        }
+                    }
+                }.resume()
+            }
+        }
+    }
+
+    /// 返回OPServer的请求的结果接口
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift:303]: in English

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+    /// - Parameters:
+    ///   - method:Method, http方法的类型
+    ///   - url:String，请求的路径
+    ///   - completion:异步返回处理好的data以及报错的类型
+    ///
+    ///  需要自己传URL类型的url过来
+    static func homeServerRequest(
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift:310]: also refactor this

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift [hunk 12]
```diff
@@ -0,0 +1,486 @@
...
+                }.resume()
+            }
+        } else {
+            completion(.failure(.responseError))
+        }
+    }
+}
+
+// String的扩展，让其具有直接加键值对的功能
+extension String {
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift:456]: Do not use this extension. Use `URLQueryItem` to build `URL` then send request instead.

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/Models/ArtifactRatingScore.swift [hunk 13]
```diff
@@ -0,0 +1,157 @@
+//
+//  ArtifactRatingScore.swift
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/Models/ArtifactRatingScore.swift:2]: delete this whole file

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/Models/ENCharacterLoc.swift [hunk 14]
```diff
@@ -0,0 +1,94 @@
+//
+//  ENCharacterLoc.swift
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/Models/ENCharacterLoc.swift:2]: delete this whole file

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/RequestRelated.swift [hunk 15]
```diff
@@ -0,0 +1,248 @@
+//
+//  RequestRelated.swift
+//
+//
+//  Created by Bill Haku on 2023/3/27.
+//
+
+import Foundation
+
+// public typealias FetchResult = Result<UserData, FetchError>
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/RequestRelated.swift:10]: delete those comment

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/RequestRelated.swift [hunk 16]
```diff
@@ -0,0 +1,248 @@
...
+// MARK: - ErrorCode
+
+public struct ErrorCode: Codable {
+    var code: Int
+    var message: String?
+}
+
+// MARK: - FetchError
+
+public enum FetchError: Error, Equatable {
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/RequestRelated.swift:53]: delete this

File: Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/RequestRelated.swift [hunk 17]
```diff
@@ -0,0 +1,248 @@
...
+}
+
+// MARK: - PSAServerError
+
+public enum PSAServerError: Error {
+    case uploadError(String)
+    case getDataError(String)
+}
+
+extension FetchError {
```
Comment (CanglongCl) [inline Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/RequestRelated.swift:93]: delete this

**Review by CanglongCl — CHANGES_REQUESTED** [review]

File: Packages/HBMihoyoAPI/Sources/HBMihoyoAPI/Common/Models/Server.swift (hunk 4 above)
Comment (Bill-Haku) [reply Packages/HBMihoyoAPI/Sources/HBMihoyoAPI/Common/Models/Server.swift:62]: i18n is not needed here. same as that in ophelper

**Review by CanglongCl — APPROVED** [review]

**Pull request merged**